src/stan/callbacks/unique_stream_writer.hpp

```diff
@@ -96,6 +96,24 @@ class unique_stream_writer final : public writer {
     *output_ << values.transpose().format(CommaInitFmt);
   }
 
+  /**
+   * Writes multiple rows and columns of values in csv format.
+   *
+   * Note: the precision of the output is determined by the settings
+   *  of the stream on construction.
+   *
+   * @param[in] values An array of values. The input is expected to have
+   * parameters in the rows and samples in the columns. The array is then
+   * transposed for the output.
+   */
+  void operator()(const Eigen::Array<double, -1, -1>& values) {
```

Comment:
This overload is now causing failures because it is ambiguous (I guess because of the implicit conversion constructor between the Array and Matrix classes?)

This may be a silly question, but because of where it is used, we never actually need the memory as an Array again, so is there some way we could move-construct a matrix and then call the existing overload?